### PR TITLE
Dev/navbar

### PR DESCRIPTION
**Pull request by Joelzac537**
### Functionality:
Breadcrumbs Indicate the current page’s location within a navigational hierarchy that automatically adds separators via CSS.

### Solution:
Need to add routes

### Risk level:
- [x] high 
- [ ] medium
- [ ] low

### How to test:
From dashboard , you can go to the create new session page and without reloading you can come back or go to any other page using breadcrumbs

### CODE REVIEW

**Review by shivamgoyal7 — APPROVED** [review]
Ready to merge

File: worktrolly-ui/src/app/app.module.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import { TaskCardsComponent } from './body/tasks/task-cards/task-cards.component
 import { TaskDetailsComponent } from './body/task-details/task-details.component';
 import { LogWorkComponent } from './body/task-details/log-work/log-work.component';
 import { EditPageComponent } from './body/task-details/edit-page/edit-page.component';
+import { BreadcrumbComponent } from './breadcrumb/breadcrumb.component';
```
Comment (shivamgoyal7) [inline worktrolly-ui/src/app/app.module.ts:23]: Remove this line

File: worktrolly-ui/src/app/app.module.ts [hunk 2]
```diff
@@ -36,6 +37,7 @@ import { EditPageComponent } from './body/task-details/edit-page/edit-page.compo
     TaskDetailsComponent,
     LogWorkComponent,
     EditPageComponent,
+    BreadcrumbComponent,
```
Comment (shivamgoyal7) [inline worktrolly-ui/src/app/app.module.ts:40]: remove breadcrumb component

**Review by shivamgoyal7 — CHANGES_REQUESTED** [review]

File: worktrolly-ui/src/app/app.module.ts (hunk 2 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/app.module.ts:40]: removed

File: worktrolly-ui/src/app/app.module.ts (hunk 1 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/app.module.ts:23]: remolved

**Review by shivamgoyal7 — APPROVED** [review]
ship it

**Comment by viv9k** [comment]
Please attach screenshots as well

**Comment by Joelzac537** [comment]
Screenshot's
WorktrollyUi - Google Chrome 16-12-2020 22_46_05
WorktrollyUi - Google Chrome 16-12-2020 22_46_17
WorktrollyUi - Google Chrome 16-12-2020 22_46_57
WorktrollyUi - Google Chrome 16-12-2020 22_47_15
WorktrollyUi - Google Chrome 16-12-2020 22_47_28
WorktrollyUi - Google Chrome 16-12-2020 22_47_48

**Comment by Joelzac537** [comment]
Screenshots 
WorktrollyUi - Google Chrome 26-12-2020 18_02_09
WorktrollyUi - Google Chrome 26-12-2020 18_02_18
WorktrollyUi - Google Chrome 26-12-2020 18_02_28
WorktrollyUi - Google Chrome 26-12-2020 18_02_50
WorktrollyUi - Google Chrome 26-12-2020 18_03_25
WorktrollyUi - Google Chrome 26-12-2020 18_03_35

**Comment by viv9k** [comment]
We can have a single component of navbar that can handle this, that will reduce lots of code

**Review by viv9k — CHANGES_REQUESTED** [review]
We can have directly new component for this feature.

**Comment by Joelzac537** [comment]
Updated screenshot
WorktrollyUi - Google Chrome 11-01-2021 08_55_45
WorktrollyUi - Google Chrome 11-01-2021 08_55_54
WorktrollyUi - Google Chrome 11-01-2021 08_56_06
WorktrollyUi - Google Chrome 11-01-2021 08_56_15
WorktrollyUi - Google Chrome 11-01-2021 08_56_25
WorktrollyUi - Google Chrome 11-01-2021 08_56_38

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.ts [hunk 3]
```diff
@@ -15,14 +16,17 @@ export class LogWorkComponent implements OnInit {
   @Input('task') task: Tasks
   @Output() logWorkCompleted = new EventEmitter<{ completed: boolean }>();
 
+  logWork: string= "/ Log Work"
+  taskDetails: string= "/ Task Details"
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/log-work/log-work.component.ts:20]: / can be handled in navbar component.

Comment (viv9k) [reply worktrolly-ui/src/app/body/task-details/log-work/log-work.component.ts:20]: Also give name to each component, like componentName = "abc".

And send this component name to navbar component.

Then in navbar component have an array of component names. That can be shown to the navbar component.
The component which is at the end will only be marked green.
Also perform push and pop operation in navbar component over the list of components that are in the arraylist

**Review by viv9k — CHANGES_REQUESTED** [review]
Also give name to each component, like componentName = "abc".

And send this component name to navbar component.

Then in navbar component have an array of component names. That can be shown to the navbar component.
The component which is at the end will only be marked green.
Also perform push and pop operation in navbar component over the list of components that are in the arraylist

File: worktrolly-ui/package.json [hunk 4]
```diff
@@ -20,13 +20,14 @@
         "@angular/platform-browser": "~11.0.2",
         "@angular/platform-browser-dynamic": "~11.0.2",
         "@angular/router": "~11.0.2",
+        "angular-crumbs": "^3.0.1",
```
Comment (viv9k) [inline worktrolly-ui/package.json]: Please revert the changes in this file

Comment (viv9k) [reply worktrolly-ui/package.json]: And also in package-lock.json

File: worktrolly-ui/src/app/body/create-new-session/create-new-session.component.spec.ts [hunk 5]
```diff
@@ -22,4 +22,4 @@ describe('CreateNewSessionComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
-});
+});
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-session/create-new-session.component.spec.ts:25]: Revert the changes in this file

File: worktrolly-ui/src/app/body/create-new-session/create-new-session.component.html [hunk 6]
```diff
@@ -1,7 +1,10 @@
-<div *ngIf="!enableLoader" class="container py-5">
-    <div id="createNewTaskContainer">
+<div class="container py-5">
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-session/create-new-session.component.html:1]: Revert the changes in this file

File: worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts [hunk 7]
```diff
@@ -17,6 +18,7 @@ export class CreateNewSprintComponent implements OnInit {
   @ViewChild('form') form: NgForm;
   @Input('newSprintId') newSprintId: string;
 
+
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts:21]: Revert the changes in this file

File: worktrolly-ui/src/app/body/task-details/edit-page/edit-page.component.ts [hunk 8]
```diff
@@ -52,7 +56,17 @@ export class EditPageComponent implements OnInit {
...
   }
+  
+  backToTasks(category: string){
+    this.sprintName = "S" + this.task.SprintNumber;
+    this.router.navigate(['Tasks', this.task.Category, this.sprintName])
+  }
+  
+  backToDashboard(){
+    this.router.navigate(['']);
+  }
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/edit-page/edit-page.component.ts:70]: Remove these changes

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html [hunk 9]
```diff
@@ -1,19 +1,8 @@
...
-                    </ul>
-                    <ul class="nav navbar-nav navbar-right">
-                        <li class="nav-item">
-                            <span type="button" class="material-icons text-light" (click)="workDone()">close</span>
-                        </li>
-                    </ul>
-                </nav>
+        <div class="container px-0 py-5" id="logWorkPage">             
+            <div class="bg-dark py-2 px-5"> 
+                <h3 class="pt-5 fontHeading text-white">Log Work</h3>
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html:5]: Revert these changes

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.ts [hunk 10]
```diff
@@ -43,6 +49,17 @@ export class LogWorkComponent implements OnInit {
...
+
+  backToTasks(category: string){
+    this.sprintName = "S" + this.task.SprintNumber;
+    this.router.navigate(['Tasks', category, this.sprintName])
+  }
+
+  backToDashboard(){
+    this.router.navigate(['']);
+  }
+
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/log-work/log-work.component.ts:64]: Remove these changes

Comment (viv9k) [reply worktrolly-ui/src/app/body/task-details/log-work/log-work.component.ts:64]: Also remove router

File: worktrolly-ui/src/app/body/task-details/task-details.component.html [hunk 11]
```diff
@@ -1,7 +1,7 @@
 <ng-container *ngIf="taskDataObservable | async">
     <ng-container *ngIf="!logWorkEnabled && !editTaskEnabled">
-        <div class="container p-5">
-            <div class="card text-dark mb-3" id="cardDescription">
+        <div class="container p-5 taskDetail">
+            <div class="card text-dark mb-3" id="cardDescription">
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/task-details.component.html:4]: Revert the changes in this file

File: worktrolly-ui/src/app/body/tasks/tasks.component.html [hunk 12]
```diff
@@ -1,12 +1,19 @@
...
+                        <li class="nav-item">
+                            <h5 class="active nav-link" style="font-weight:bolder">Tasks</h5>
+                        </li>
+                    </ul>
+                    <ul class="nav navbar-nav navbar-right">
+                        <li class="nav-item">
+                            <span type="button" class="material-icons text-light" (click)="backToDashboard()">close</span>
+                        </li>
+                    </ul>
+                </nav>
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/tasks/tasks.component.html:16]: revert the changes done in this file

File: worktrolly-ui/src/app/services/navbar-holder.service.ts [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+      } else {
+        this.currentNav = "";
+      }
+     
+    }
+
+  }
+
+}
+
```
Comment (viv9k) [inline worktrolly-ui/src/app/services/navbar-holder.service.ts:54]: Remove unnecessary blank spaces

**Review by viv9k — CHANGES_REQUESTED** [review]
Too many rubbish changes fix them asap

File: worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts (hunk 7 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts:21]: reverted

File: worktrolly-ui/src/app/services/navbar-holder.service.ts (hunk 13 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/services/navbar-holder.service.ts:54]: removed

File: worktrolly-ui/src/app/body/tasks/tasks.component.html (hunk 12 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/tasks/tasks.component.html:16]: changed

File: worktrolly-ui/src/app/body/task-details/task-details.component.html (hunk 11 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/task-details/task-details.component.html:4]: changed

File: worktrolly-ui/src/app/body/create-new-session/create-new-session.component.spec.ts (hunk 5 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/create-new-session/create-new-session.component.spec.ts:25]: no change needed here

File: worktrolly-ui/package.json (hunk 4 above)
Comment (Joelzac537) [reply worktrolly-ui/package.json]: already removed

File: worktrolly-ui/src/app/body/create-new-session/create-new-session.component.html (hunk 6 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/create-new-session/create-new-session.component.html:1]: changed

File: worktrolly-ui/src/app/body/task-details/edit-page/edit-page.component.ts (hunk 8 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/task-details/edit-page/edit-page.component.ts:70]: removed

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html (hunk 9 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html:5]: changed

File: worktrolly-ui/package.json [hunk 14]
```diff
@@ -54,4 +54,4 @@
         "tslint": "~6.1.0",
         "typescript": "~4.0.2"
     }
-}
```
Comment (viv9k) [inline worktrolly-ui/package.json:57]: Revert this file

File: worktrolly-ui/src/app/app-routing.module.ts [hunk 15]
```diff
@@ -15,7 +15,7 @@ const redirectLoggedInToDashboard = () => redirectLoggedInTo(['']);
 
 const routes: Routes = [
   { path: 'Board', component: BoardComponent },
-  { path: 'CreateNewSession', component: CreateNewSessionComponent },
```
Comment (viv9k) [inline worktrolly-ui/src/app/app-routing.module.ts:18]: revert this file

File: worktrolly-ui/src/app/body/create-new-session/create-new-session.component.ts [hunk 16]
```diff
@@ -30,6 +31,7 @@ export class CreateNewSessionComponent implements OnInit {
   constructor(private functions: AngularFireFunctions, private router: Router, private location: Location) { }
 
   ngOnInit(): void {
+
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-session/create-new-session.component.ts:34]: revert this file

File: worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.html [hunk 17]
```diff
@@ -1,5 +1,5 @@
-<div *ngIf="!enableLoader" class="container">
-    <div class="mt-5" id="startNewSprintContainer">
+<div class="container">
+    <div class="mt-5 px-5 py-5" id="startNewSprint">
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.html:2]: revert this file

File: worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts [hunk 18]
```diff
@@ -107,11 +108,9 @@ export class CreateNewSprintComponent implements OnInit {
       this.enableLoader = false;
       console.error("Error", error);
     }
-
   }
-
+
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts:112]: revert this file

File: worktrolly-ui/src/app/body/task-details/edit-page/edit-page.component.html [hunk 19]
```diff
@@ -1,6 +1,8 @@
+<form (ngSubmit)="editPage()" #form="ngForm">
+    <div class="container py-5" id="editPageTask">
 <form *ngIf="!enableLoader" (ngSubmit)="editPage()" #form="ngForm">
     <div class="container p-3">
-        <div class="card text-dark mb-3" id="editPageTaskContainer">
+        <div class="card text-dark mb-3" id="editPageTaskContainer">
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/edit-page/edit-page.component.html:5]: revert this file

File: worktrolly-ui/src/app/body/task-details/edit-page/edit-page.component.ts [hunk 20]
```diff
@@ -15,13 +16,16 @@ export class EditPageComponent implements OnInit {
...
   @Output() editTaskCompleted = new EventEmitter<{ completed: boolean }>();
 
+  componentName: string = "Edit Task"
   editTask: Tasks
   previousSprintId: number
+  sprintName: string
   enableLoader: boolean = false
 
-  constructor(private functions: AngularFireFunctions) { }
+  constructor(private functions: AngularFireFunctions, private router: Router, private navbarHolder: NavbarHolderService) { }
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/edit-page/edit-page.component.ts:25]: remove router from this component

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html [hunk 21]
```diff
@@ -14,21 +14,10 @@ <h5 class="active nav-link" style="font-weight:bolder">Log Work</h5>
...
-                    <div class="alert alert-info form-inline col-md-5" role="alert">
-                        <div style="font-weight: bold;" class="pr-3">Sprint Number:</div>
-                        <div id="logSprintNumber">{{SprintName}}</div>
-                    </div>
-                </div>
                 <div class="alert alert-info form-inline col-md-12" role="alert">
                     <div style="font-weight: bold;" class="pr-3">Title: </div>
                     <div>{{task.Title}}</div>
-                </div> -->
+                </div>
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html:20]: revert this file

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.ts [hunk 22]
```diff
@@ -14,17 +16,21 @@ export class LogWorkComponent implements OnInit {
...
   Id: string
   logWorkDone: number
   logWorkStatus: number
   logHours: number
   logWorkComment: number
+  sprintName: string
   enableLoader: boolean = false
 
-  constructor(private functions: AngularFireFunctions) { }
+  constructor(private functions: AngularFireFunctions,private router: Router,private navbarHolder: NavbarHolderService) { }
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/log-work/log-work.component.ts:29]: remove router from this component

File: worktrolly-ui/src/app/body/tasks/tasks.component.ts [hunk 23]
```diff
@@ -26,13 +27,17 @@ export class TasksComponent implements OnInit {
   filterStatus: string
   filterCategory: string
   showFilter: boolean = false
-  constructor(private route: ActivatedRoute, private router: Router, private db: AngularFirestore) { }
+  constructor(private route: ActivatedRoute, private router: Router, private db: AngularFirestore, private navbarHolder: NavbarHolderService) { }
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/tasks/tasks.component.ts:30]: route is not needed here

**Review by viv9k — CHANGES_REQUESTED** [review]
Remove all the unnecessary changes

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html (hunk 21 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html:20]: code is correct now

**Comment by viv9k** [comment]
Please resolve merge conflict

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html (hunk 21 above)
Comment (viv9k) [reply worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html:20]: Please revert this file directly

File: worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts (hunk 18 above)
Comment (viv9k) [reply worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts:127]: Please revert this file

File: worktrolly-ui/src/app/body/create-new-session/create-new-session.component.html [hunk 24]
```diff
@@ -1,7 +1,10 @@
 <div *ngIf="!enableLoader" class="container py-5">
     <div id="createNewTaskContainer">
+        <div class="row pt-2">
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-session/create-new-session.component.html:3]: Revert this file

File: worktrolly-ui/package.json (hunk 14 above)
Comment (viv9k) [reply worktrolly-ui/package.json:57]: Revert this file

**Review by viv9k — CHANGES_REQUESTED** [review]
Yet there are many nonsense changes.

File: worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts (hunk 18 above)
Comment (Joelzac537) [reply worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts:127]: reverted

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html (hunk 21 above)
Comment (viv9k) [reply worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html:20]: Please revert this file

**Review by viv9k — CHANGES_REQUESTED** [review]
Please do the required changes asap

File: worktrolly-ui/src/app/body/create-new-session/create-new-session.component.html [hunk 25]
```diff
@@ -1,7 +1,10 @@
 <div *ngIf="!enableLoader" class="container py-5">
     <div id="createNewTaskContainer">
         <div class="row">
-            <div class="col">
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-session/create-new-session.component.html:4]: restore this file

File: worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts [hunk 26]
```diff
@@ -109,25 +109,21 @@ export class CreateNewSprintComponent implements OnInit {
...
       this.enableLoader = false;
       console.error("Error", error);
     }
-
   }
 
   backToDashboard(){
     this.location.back();
   }
-
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts:132]: Restore this file

File: worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html [hunk 27]
```diff
@@ -24,17 +24,15 @@ <h5 class="active nav-link" style="font-weight:bolder">Log Work</h5>
                         <div style="font-weight: bold;" class="pr-3">Sprint Number:</div>
                         <div id="logSprintNumber">{{SprintName}}</div>
                     </div>
-                </div>
+                 </div> -->
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/task-details/log-work/log-work.component.html:27]: Restore this file

**Review by viv9k — CHANGES_REQUESTED** [review]
restore the requested file

File: worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts (hunk 26 above)
Comment (viv9k) [reply worktrolly-ui/src/app/body/create-new-sprint/create-new-sprint.component.ts:132]: I don't see this resolved

**Review by viv9k — CHANGES_REQUESTED** [review]
complete the change

**Comment by viv9k** [comment]
I have written the code merged it into dev-angular branch, so this pr can be closed

**Pull request closed without merging**